src/transaction/TransactionController.ts

```diff
@@ -495,9 +500,11 @@ export class TransactionController extends BaseController<
     };
 
     try {
-      const { gas, gasPrice } = await this.estimateGas(transaction);
+      const { gas } = await this.estimateGas(transaction);
       transaction.gas = gas;
-      transaction.gasPrice = gasPrice;
+
+      // TODO(eip1559) check if this is not needed for legacy
```

Comment:
Does this `TODO` block the PR from landing?

Reply:
before we were setting the gasPrice if it didn’t exist when adding a transaction, but we've removed that.

I've removed the comment now.